gremlin-go/driver/client.go

```diff
@@ -71,19 +78,64 @@ func NewClient(url string, configurations ...func(settings *ClientSettings)) (*C
 		transporterType: settings.TransporterType,
 		connection:      conn,
 	}
+	// TODO: PoolSize must be 1 on session mode. Implement after AN-980
```

Comment:
Where is this from? Not saying it doesn't make sense but I don't see this lock in gremlin-python

Reply:
Gremlin-Python has this block in the client constructor
```python
        if self._session_enabled:
            if pool_size is None:
                pool_size = 1
            elif pool_size != 1:
                raise Exception("PoolSize must be 1 on session mode!")
```